### PR TITLE
Proposers rewards

### PR DESCRIPTION
**Pull request by danielSanchezQ**
Need of #1 to be merged first.

### CODE REVIEW

**Comment by danielSanchezQ** [comment]
Since it was implemented we can already use it. But most probably I'll migrate it to rust and reuse things from `vit-servicing-station-lib` that would be much handier.

File: README.md [hunk 1]
```diff
@@ -59,3 +59,41 @@ OPTIONS:
...
+
+  Calculate catalyst rewards after tallying process. If all --proposals-
+  path, --active-voteplan-path and --challenges_path are provided data is
+  loaded from the json files on those locations. Otherwise data is requested
+  to the proper API endpoints pointed to the --vit-station-url option.
+
+Options:
+  --conversion-factor FLOAT       [required]
+  --output-file TEXT              [required]
+  --threshold FLOAT               [default: 0.15]
```
Comment (Zeegomo) [inline README.md:83]: Nit: maybe a more descriptive name like approval threshold?

File: scripts/python/proposers_rewards.py [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+# File loaders
+
+def load_json_from_file(file_path: str) -> Dict:
+    with open(file_path, encoding="utf-8") as f:
+        return json.load(f)
+
+
+def get_proposals_from_file(proposals_file_path: str) -> Dict[str, Proposal]:
+    proposals: Generator[Proposal, None, None] = (
+        Proposal(**proposal_data) for proposal_data in load_json_from_file(proposals_file_path)
```
Comment (Zeegomo) [inline scripts/python/proposers_rewards.py:121]: what is the advantage of using a generator instead of list comprehension?

**Review by Zeegomo — APPROVED** [review]
LGTM

File: scripts/python/proposers_rewards.py (hunk 2 above)
Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:121]: It is not consumed immediately which avoids the intermediate allocation.

File: README.md (hunk 1 above)
Comment (danielSanchezQ) [reply README.md:83]: Definitely, thanks :)

File: scripts/python/proposers_rewards.py (hunk 2 above)
Comment (Zeegomo) [reply scripts/python/proposers_rewards.py:121]: ahh, so also the dict below is created piece by piece upon request?

Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:121]: Yeah, generators are lazy so it is consumed by the dict below.

File: scripts/python/proposers_rewards.py [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+class DecryptedTally(pydantic.BaseModel):
+    decrypted: TallyResult
+
+
+class PrivateTallyState(pydantic.BaseModel):
+    state: DecryptedTally
+
+
+class PrivateTally(pydantic.BaseModel):
+    Private: PrivateTallyState
```
Comment (enzoc4) [inline scripts/python/proposers_rewards.py:66]: just curious, why is this uppercase? (`Private` instead of `private`)

**Review by enzoc4 — APPROVED** [review]

File: scripts/python/proposers_rewards.py [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+        proposals_result = await client.get(f"{vit_servicing_station_url}/api/v0/proposals")
+        assert proposals_result.status_code == 200
+        return [Proposal(**proposal_data) for proposal_data in proposals_result.json()]
+
+
+async def get_active_voteplans_from_api(vit_servicing_station_url: str) -> List[VoteplanStatus]:
+    async with httpx.AsyncClient() as client:
+        proposals_result = await client.get(f"{vit_servicing_station_url}/api/v0/vote/active/plans")
+        assert proposals_result.status_code == 200
+        return [VoteplanStatus(**proposal_data) for proposal_data in proposals_result.json()]
```
[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:173]: Yeah, they can be refactored. I thought about it too, Ill do.

File: scripts/python/proposers_rewards.py (hunk 3 above)
Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:66]: in the talley result it is defined as uppercase:

```json
"tally": {
          "Public": {
```

**Pull request merged**